app/src/features/mappings/MappingsTable.tsx

```diff
@@ -0,0 +1,42 @@
+import React from "react";
+
+import { CircularProgress, Table, TableBody } from "@material-ui/core";
+import { useParams } from "react-router";
+
+import TableContainer from "common/TableContainer/TableContainer";
+import {
+  useListSourceResources,
+  useRetrieveSourceQuery,
+} from "services/api/api";
+
+import MappingRow from "./MappingRow";
+
+const MappingsTable = (): JSX.Element => {
+  const { sourceId } = useParams<{ sourceId?: string }>();
+
+  if (!sourceId) {
+    return <></>;
+  }
+
+  const { data: source } = useRetrieveSourceQuery({ id: sourceId ?? "" });
```

Comment:
I think you can't do that: using a hook inside a conditional (since you have `if (!sourceId) return <></>` above). https://flaviocopes.com/react-hooks-conditionals/ https://fr.reactjs.org/docs/hooks-rules.html